app/src/usb.c

```diff
@@ -33,11 +33,8 @@ enum usb_dc_status_code zmk_usb_get_status(void) { return usb_status; }
 enum zmk_usb_conn_state zmk_usb_get_conn_state(void) {
     LOG_DBG("state: %d", usb_status);
     switch (usb_status) {
-    case USB_DC_SUSPEND:
     case USB_DC_CONFIGURED:
     case USB_DC_RESUME:
-    case USB_DC_CLEAR_HALT:
-    case USB_DC_SOF:
```

Comment:
Have you had a chance to confirm the state it enters when a host is connected but sleeps/requests the USB device sleep? We should also confirm this is set up properly when making changes here.

Also, are we sure this is consistent across *all* the Zephyr USB drivers?

Reply:
Hi, I tested these additional scenarios:
1. Connecting the keyboard to my laptop, sleeping the laptop, and waking it via keyboard, this works
2. Sleeping the laptop, then connect the keyboard, then wake it via keyboard, this works as well

Strangely, the keyboard didn't transition to any suspended state with the laptop, which feels concerning. If it helps, the board I'm testing with is nice nano v2, but I'm uncertain if this behaviour should be expected across all USB drivers.

To account for other behaviors from other devices, perhaps a better approach is to revert the mapping changes, and keep track that the keyboard has at least gone through the `USB_DC_CONFIGURED` state once, before allowing it as a possible endpoint? (Need to see if this works with `ZMK_USB_BOOT`)

Reply:
> 
> I was being dumb and locked my laptop instead of sleeping it properly, the state does get set into suspended. When suspended, I wasn't able to wake the laptop from the keyboard, but I want to also check if this is expected behavior even from `main` branch.

It *should*, if you have `CONFIG_USB_DEVICE_REMOTE_WAKEUP=y` and the USB driver supports it (which the nrfx one does). In particular, since "suspended" is still considered USB active, then any attempt to send will still pick the USB endpoint, which ultimately triggers this code: https://github.com/zmkfirmware/zmk/blob/main/app/src/usb_hid.c#L132


> 
> To account for other behaviors from other devices, perhaps a better approach is to revert the mapping changes, and keep track that the keyboard has at least gone through the `USB_DC_CONFIGURED` state once, before allowing it as a possible endpoint? (Need to see if this works with `ZMK_USB_BOOT`)

Yeah, I think "has gone through configured state but hasn't gone to disconnected" may be the best way to really track this?

Reply:
Thanks for the pointers! I changed the code to what we mentioned above, and regression tested that both the wake and boot protocol functionality still works as expected, and also updated the PR description to match.

I did it by adding another `is_configured` bool, but if space is a concern, it's possible to change `usb_status` to store both the status code and this flag as a bitfield and parsing out the values separately, at the cost of readability.